lib/Cake/Model/Datasource/DboSource.php

```diff
@@ -345,6 +345,11 @@ public function value($data, $column = null, $null = true) {
 			case 'boolean':
 				return $this->_connection->quote($this->boolean($data, true), PDO::PARAM_BOOL);
 			case 'string':
+				if ($data === '') {
+					return 'NULL';
```

Comment:
Why is this necessary for UUID support? Couldn't the UUID requirements be added as a case in this switch?